src/comments.js

```diff
@@ -870,15 +874,14 @@ function printComments(comments, options) {
   comments.forEach((comment, index, comments) => {
     comment.printed = true;
     parts.push(comment.value);
-    parts.push(hardline);
     if (
       isNextLineEmpty(options.originalText, comment, options) &&
       comments.length > index + 1
     ) {
       parts.push(hardline);
     }
   });
-  return concat(parts);
+  return join(hardline, parts);
```

Comment:
I think here problem with extra hard lines, why we need hardline here?

Reply:
Comments are now attached to `noop` nodes, so trailing newlines are printed automatically - all we need is to make sure newline between the parts are still printed.

Reply:
:+1: